src/ray/util/logging.cc

```diff
@@ -189,23 +188,22 @@ void RayLog::StartRayLog(const std::string &app_name, RayLogLevel severity_thres
         app_name_without_path = app_file_name;
       }
     }
-    char buffer[80];
-    time_t rawtime;
-    time(&rawtime);
-#ifdef _WIN32
-    int pid = _getpid();
-#else
-    pid_t pid = getpid();
-#endif
-    strftime(buffer, sizeof(buffer), "%Y%m%d-%H%M%S", localtime(&rawtime));
-    std::string path = dir_ends_with_slash + app_name_without_path + "." + buffer + "." +
-                       std::to_string(pid) + ".log";
-    stream_logger_singleton.out_.rdbuf()->pubsetbuf(0, 0);
-    stream_logger_singleton.out_.open(path.c_str(),
-                                      std::ios_base::app | std::ios_base::binary);
+    app_name_without_path += ".";
+    google::SetLogFilenameExtension(app_name_without_path.c_str());
+    google::SetLogDestination(level, dir_ends_with_slash.c_str());
+    FLAGS_stop_logging_if_full_disk = true;
+  } else {
+    // NOTE(lingxuan.zlx): If no specific log dir or empty directory string,
+    // we use stdout by default.
+    google::base::SetLogger(level, &stdout_logger_singleton);
   }
-  for (int lvl = 0; lvl < NUM_SEVERITIES; ++lvl) {
-    google::base::SetLogger(lvl, &stream_logger_singleton);
+  for (int i = static_cast<int>(RayLogLevel::DEBUG);
+       i <= static_cast<int>(RayLogLevel::FATAL); ++i) {
+    if (i != level) {
+      // NOTE(lingxuan.zlx): It means nothing can be printed or sinked to pass
+      // an empty destination.
+      google::SetLogDestination(i, "");
```

Comment:
Could you put this link to the comment above? It's not obvious that an empty string means don't write. https://github.com/google/glog/blob/0a2e5931bd5ff22fd3bf8999eb8ce776f159cda6/src/logging.cc#L1110